ember-addon-main.js

```diff
@@ -18,17 +19,21 @@ module.exports = {
   },
 
   setupPreprocessorRegistry: function(type, registry) {
-    var self = this;
-
     // ensure that broccoli-ember-hbs-template-compiler is not processing hbs files
     registry.remove('template', 'broccoli-ember-hbs-template-compiler');
 
+    var htmlbarsOptions = this.htmlbarsOptions();
```

Comment:
This needs to be lazy because we need for the full `setupPreprocessorRegistry` phase to be finished before we attempt to "realize" the known plugins (we cannot guarantee that every addon that will register a plugin will have done it before our `setupPreprocessorRegistry`).

tldr; we can't memoize `this.htmlbarsOptions()`, it needs to be called lazily in the `precompile` and `toTree` methods.